chore: Microsoft email channel integration (#376)
chatwoot/docs: docs/self-hosted/configuration/features/email-channel/azure-app-setup.md

### Going into production.

Before you can start using your Azure app in production, you will have to get it verified by Azure. Refer to the [docs](https://learn.microsoft.com/en-us/azure/active-directory/develop/howto-configure-publisher-domain) on getting your app verified. And learn more about publisher verification [here](https://learn.microsoft.com/en-us/azure/active-directory/develop/publisher-verification-overview)

1. Go to: `Branding & Properties` and add your domain details, and then click `Verify and Save.`

![verify_publisher](./images/azure/verify-publisher.png)

We are handling the mentioned steps about the hosting  `microsoft-identity-association.json`.
Once you set `AZURE_APP_ID` in your environment variable, consider this step is completed.

### Next steps

You're done! Next, you should [enable the outlook email channel](https://www.chatwoot.com/docs/product/channels/email/microsoft/create-channel) in chatwoot inbox.

Consider `test-imap@outlook.com` is an email for which you are adding a new inbox.

![configuration](../../../../product/channels/email/microsoft/images/ms-oauth-email-config.gif)

### NOTE

> If you don't get your publisher verified you will be able to attach and fetch mails from azure app registration email.

Reviewer: ```suggestion
You need to get your publisher verified to be able to attach and fetch mails from the Azure app registration email.
```

Reviewer: What do you intend here? Could you rephrase if this is not what you want to say?

Author: It is mentioned in the `### Next step` I just added this in `Note` again to ensure the reader follows the verification process. :))
